api/envoy/api/v2/route/route.proto

```diff
@@ -846,6 +858,24 @@ message RetryPolicy {
   repeated uint32 retriable_status_codes = 7;
 }
 
+// HTTP request hedging TODO(mpuncel) docs
+// [#not-implemented-hide:]
+message HedgePolicy {
+  // Specifies the number of initial requests that should be sent upstream.
+  // Must be at least 1. If the number is fractional, the number of requests
+  // sent will be at least the whole portion of the number, with the remainder
+  // used to decide probabilistically whether to send another request. For
+  // example, a value of 2.3 means that 2 requests will be sent 70% of the
+  // time, and 3 requests otherwise.
+  // Defaults to 1.
+  google.protobuf.FloatValue initial_requests = 1;
```

Comment:
I don't feel strongly about it, but this might be a little easier to implement if you:
1) Have a fixed initial request value
2) and an optional https://github.com/envoyproxy/envoy/blob/master/api/envoy/type/percent.proto#L27 (FractionalPercent) for an extra request.

I would also put a validator on the initial requests to make sure it's at least 1.

Reply:
sounds good, it'll probably be easier for people to understand too